[BUG] Messages inside room disappears after page reload

## 🐛 Bug Report

**Describe the bug**
The messages sent and received inside the room disappears once the page is reloaded.

**To Reproduce**

Steps to reproduce the behavior:
1. Create a room and join
2. Send a message
3. Reload the page
4. Message dissappears

**Expected behavior**

After page reload the message should not be disappeared.

**Screenshots**

### Before Reload with message sent:
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/c126a1f8-ff84-4d28-a730-459f88db3f19" />

### After Reload:
<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/7f782424-428e-48ea-85ed-f3be8ca5bdf7" />

**Environment (please complete the following information):**
- OS: Windows 11
- Browser: Chrome 
- Node.js version: 22+
- Device: Desktop

@Condition00 
i am getting this error when trying to run client on my local


`npm run dev

> dialectica-ai-client@1.0.0 dev
> next dev -p 3001

   ▲ Next.js 15.3.5
   - Local:        http://localhost:3001
   - Network:      http://192.168.1.30:3001

 ✓ Starting...
 ✓ Ready in 1146ms
 ○ Compiling / ...
 ⨯ src/app/layout.tsx
An error occurred in `next/font`.

Error: Cannot find module '../lightningcss.darwin-arm64.node'
Require stack:
- /Users/nehamate/Documents/Repos/dialectica-ai/client/node_modules/lightningcss/node/index.js
- /Users/nehamate/Documents/Repos/dialectica-ai/client/node_modules/@tailwindcss/node/dist/index.js
- /Users/nehamate/Documents/Repos/dialectica-ai/client/node_modules/@tailwindcss/postcss/dist/index.js
- /Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/build/webpack/config/blocks/css/plugins.js
- /Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/build/webpack/config/blocks/css/index.js
- /Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/build/webpack/config/index.js
- /Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/build/webpack-config.js
- /Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/server/dev/hot-reloader-webpack.js
- /Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/server/lib/router-utils/setup-dev-bundler.js
- /Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/server/lib/router-server.js
- /Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/server/lib/start-server.js
    at Module.<anonymous> (node:internal/modules/cjs/loader:1410:15)
    at /Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/server/require-hook.js:55:36
    at defaultResolveImpl (node:internal/modules/cjs/loader:1051:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1056:22)
    at Module._load (node:internal/modules/cjs/loader:1219:37)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:238:24)
    at Module.<anonymous> (node:internal/modules/cjs/loader:1493:12)
    at mod.require (/Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/server/require-hook.js:65:28)
    at require (node:internal/modules/helpers:152:16)
    at Object.<anonymous> (/Users/nehamate/Documents/Repos/dialectica-ai/client/node_modules/lightningcss/node/index.js:22:22)
    at Module._compile (node:internal/modules/cjs/loader:1738:14)
    at Object..js (node:internal/modules/cjs/loader:1871:10)
    at Module.load (node:internal/modules/cjs/loader:1470:32)
    at Module._load (node:internal/modules/cjs/loader:1290:12)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:238:24)
    at Module.<anonymous> (node:internal/modules/cjs/loader:1493:12)
    at mod.require (/Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/server/require-hook.js:65:28)
    at require (node:internal/modules/helpers:152:16)
    at Object.<anonymous> (/Users/nehamate/Documents/Repos/dialectica-ai/client/node_modules/@tailwindcss/node/dist/index.js:15:188)
    at Module._compile (node:internal/modules/cjs/loader:1738:14)
    at Object..js (node:internal/modules/cjs/loader:1871:10)
    at Module.load (node:internal/modules/cjs/loader:1470:32)
    at Module._load (node:internal/modules/cjs/loader:1290:12)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:238:24)
    at Module.<anonymous> (node:internal/modules/cjs/loader:1493:12)
    at mod.require (/Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/server/require-hook.js:65:28)
    at require (node:internal/modules/helpers:152:16)
    at Object.<anonymous> (/Users/nehamate/Documents/Repos/dialectica-ai/client/node_modules/@tailwindcss/postcss/dist/index.js:1:1328)
    at Module._compile (node:internal/modules/cjs/loader:1738:14)
    at Object..js (node:internal/modules/cjs/loader:1871:10)
    at Module.load (node:internal/modules/cjs/loader:1470:32)
    at Module._load (node:internal/modules/cjs/loader:1290:12)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:238:24)
    at Module.<anonymous> (node:internal/modules/cjs/loader:1493:12)
    at mod.require (/Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/server/require-hook.js:65:28)
    at require (node:internal/modules/helpers:152:16)
    at /Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/build/webpack/config/blocks/css/plugins.js:64:48
    at plugin (/Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/build/webpack/config/blocks/css/plugins.js:30:44)
    at Processor.normalize (/Users/nehamate/Documents/Repos/dialectica-ai/node_modules/postcss/lib/processor.js:18:13)
    at new Processor (/Users/nehamate/Documents/Repos/dialectica-ai/node_modules/postcss/lib/processor.js:11:25)
    at postcss (/Users/nehamate/Documents/Repos/dialectica-ai/node_modules/postcss/lib/postcss.js:26:10)
    at /Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/build/webpack/config/blocks/css/index.js:128:37
    at async /Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/build/webpack/loaders/next-font-loader/index.js:94:33
    at async Span.traceAsyncFn (/Users/nehamate/Documents/Repos/dialectica-ai/node_modules/next/dist/trace/trace.js:157:20)
 ⨯ src/app/layout.tsx
An error occurred in `next/font`.`

please help me with this.
I tried cleaning and reinstalling all the dependencies but still getting same error

--- a/client/src/app/debate/[roomId]/page.tsx
+++ b/client/src/app/debate/[roomId]/page.tsx
@@ -31,20 +31,49 @@ const ChatPage: React.FC<ChatPageProps> = () => {
     const [hasAddedWelcomeMessage, setHasAddedWelcomeMessage] = useState(false);
 
 
-    const socketRef = useRef<SocketIOClient.Socket | null>(null);
+    const socketRef = useRef<any>(null);
 
     // Join room on mount:
     useEffect(() => {
         if (roomId) {
             socketRef.current = getSocket();
 
-            // Check if already connected
+            // Generate or retrieve persistent user ID
+            let persistentUserId = localStorage.getItem('dialectica_user_id');
+            if (!persistentUserId) {
+                persistentUserId = `user_${Math.random().toString(36).substring(2, 11)}`;
+                localStorage.setItem('dialectica_user_id', persistentUserId);
+                console.log('🆔 [Client] Generated new persistent userId:', persistentUserId);
+            } else {
+                console.log('🆔 [Client] Retrieved persistent userId:', persistentUserId);
+            }
+            
+            // Set current user ID to persistent ID
+            setCurrentUserId(persistentUserId);
+
+            // Identify user to server with persistent ID
             if (socketRef.current.connected) {
-                setCurrentUserId(socketRef.current.id || "");
+                socketRef.current.emit('identify', persistentUserId);
                 setIsConnected(true);
             }
 
             // event listeners, Chat and System messages ===>
+            const historyHandler = (
+                history: Array<{ sender: string; content: string; timestamp: number; role?: string }>
+            ) => {
+                console.log('📥 [Client] Received room-history:', history?.length || 0, 'messages', history);
+                setMessages(
+                    (history || []).map((msg) => ({
+                        user: msg.sender,
+                        sender: msg.sender,
+                        content: msg.content,
+                        timestamp: msg.timestamp || Date.now(),
+                        type: "chat" as const,
+                        role: msg.role as "pro" | "con" | undefined,
+                    }))
+                );
+                console.log('✅ [Client] Messages state updated with history');
+            };
             const chatHandler = (msg: { sender: string; content: string; timestamp: number; role?: string }) => {
                 setMessages(prev => [
                     ...prev,
@@ -72,6 +101,8 @@ const ChatPage: React.FC<ChatPageProps> = () => {
                 ]);
             };
 
+            console.log('🔌 [Client] Registering socket listeners for room:', roomId);
+            socketRef.current.on("room-history", historyHandler);
             socketRef.current.on("chat-message", chatHandler);
             socketRef.current.on("system-message", systemHandler);
 
@@ -80,7 +111,13 @@ const ChatPage: React.FC<ChatPageProps> = () => {
 
             socketRef.current.on("connect", () => {
                 setIsConnected(true);
-                setCurrentUserId(socketRef.current?.id || "");
+                // On reconnect, re-identify with persistent user ID
+                const persistentUserId = localStorage.getItem('dialectica_user_id');
+                if (persistentUserId && socketRef.current) {
+                    socketRef.current.emit('identify', persistentUserId);
+                    setCurrentUserId(persistentUserId);
+                    console.log('🔄 [Client] Reconnected and re-identified as:', persistentUserId);
+                }
             });
 
             socketRef.current.on("disconnect", () => {
@@ -92,6 +129,7 @@ const ChatPage: React.FC<ChatPageProps> = () => {
             const fromRandom = urlParams.get('fromRandom');
 
             if (fromRandom && !hasAddedWelcomeMessage) {
+                console.log('🎲 [Client] Coming from random room, skipping joinRoom emit');
                 // For random room users, send a welcome message since they're already in the room
                 setMessages(prev => [
                     ...prev,
@@ -111,11 +149,13 @@ const ChatPage: React.FC<ChatPageProps> = () => {
                 window.history.replaceState({}, '', newUrl);
             } else if (!fromRandom) {
                 // calling joinRoom if NOT coming from random room
+                console.log('🚪 [Client] Emitting joinRoom for:', roomId);
                 socketRef.current.emit("joinRoom", roomId);
             }
 
             return () => {
                 if (socketRef.current) {
+                    socketRef.current.off("room-history", historyHandler);
                     socketRef.current.off("chat-message", chatHandler);
                     socketRef.current.off("system-message", systemHandler);
                     socketRef.current.off("connect");
@@ -163,8 +203,16 @@ const ChatPage: React.FC<ChatPageProps> = () => {
     // Functions to handle sending messages:
     const handleSend = () => {
         if (messageInput.trim() && roomId && socketRef.current && isConnected) {
+            console.log('📤 [Client] Sending message to room:', roomId, 'Message:', messageInput);
             socketRef.current.emit("sendMessage", messageInput, roomId);
             setMessageInput("");
+        } else {
+            console.warn('⚠️ [Client] Cannot send message:', {
+                hasInput: !!messageInput.trim(),
+                hasRoomId: !!roomId,
+                hasSocket: !!socketRef.current,
+                isConnected
+            });
         }
     };
 

--- a/server/src/index.ts
+++ b/server/src/index.ts
@@ -25,6 +25,14 @@ app.get('/', (req, res) => {
 
 const rooms: { [roomId: string]: Set<string> } = {};
 
+// Map socket.id to persistent userId
+const socketToUser: { [socketId: string]: string } = {};
+
+// In-memory message history per room (non-persistent)
+const roomMessages: {
+    [roomId: string]: Array<{ sender: string; content: string; timestamp: number; role?: string }>
+} = {};
+
 const MAX_ROOM_CAPACITY = 2;
 
 // statistics:
@@ -38,6 +46,12 @@ io.on('connection', (socket: Socket) => {
     console.log(`New client connected: ${socket.id}`);
     console.log('Current rooms:', getRoomStats());
 
+    // Handle user identification (persistent across reconnects)
+    socket.on('identify', (userId: string) => {
+        socketToUser[socket.id] = userId;
+        console.log(`🆔 User identified: socket ${socket.id} -> userId ${userId}`);
+    });
+
     socket.on('requestRandomRoom', () => {
         // Remove user from any existing rooms first
         for (const [roomId, participants] of Object.entries(rooms)) {
@@ -48,10 +62,10 @@ io.on('connection', (socket: Socket) => {
 
                 // ok so removed the socket.ids for privacy and security
 
-                // Clean up empty rooms
+                // Clean up empty rooms (but keep message history)
                 if (participants.size === 0) {
                     delete rooms[roomId];
-                    console.log(`Deleted empty room: ${roomId}`);
+                    console.log(`Deleted empty room: ${roomId} (keeping message history)`);
                 }
             }
         }
@@ -78,6 +92,10 @@ io.on('connection', (socket: Socket) => {
         socket.join(availableRoomId);
         rooms[availableRoomId].add(socket.id);
         console.log(`User ${socket.id} joined room: ${availableRoomId} (${rooms[availableRoomId].size}/${MAX_ROOM_CAPACITY})`);
+
+        // Send existing room history to the user who joined
+        console.log(`📤 [requestRandomRoom] Sending room history for ${availableRoomId}:`, roomMessages[availableRoomId]?.length || 0, 'messages');
+        socket.emit('room-history', roomMessages[availableRoomId] || []);
 
         socket.emit('randomRoomFound', availableRoomId);
        // added this to omit the first user join message because it was pointless
@@ -123,7 +141,7 @@ io.on('connection', (socket: Socket) => {
 
                 if (participants.size === 0) {
                     delete rooms[roomId];
-                    console.log(`Deleted empty room: ${roomId}`);
+                    console.log(`Deleted empty room: ${roomId} (keeping message history)`);
                 }
             }
         }
@@ -145,6 +163,10 @@ io.on('connection', (socket: Socket) => {
         rooms[room].add(socket.id);
         console.log(`Client ${socket.id} joined room: ${room} (${rooms[room].size}/${MAX_ROOM_CAPACITY})`);
 
+        // Send existing room history to the user who joined
+        console.log(`📤 [joinRoom] Sending room history for ${room}:`, roomMessages[room]?.length || 0, 'messages');
+        socket.emit('room-history', roomMessages[room] || []);
+
         // Send welcome message to the user who joined
         socket.emit('system-message', `You joined room: ${room}`);
 
@@ -156,18 +178,33 @@ io.on('connection', (socket: Socket) => {
 
     // sendMessage event ===>
     socket.on('sendMessage', (message: string, room: string) => {
+        console.log(`📨 [Server] Received sendMessage from ${socket.id} - Room: ${room}, Message: ${message}`);
+        // Use persistent userId if available, otherwise fall back to socket.id
+        const senderId = socketToUser[socket.id] || socket.id;
         const messageData = {
-            sender: socket.id,
+            sender: senderId,
             content: message,
             timestamp: Date.now(),
             role: undefined // later for pro/con roles
         };
+        // Persist in room history with a rolling cap
+        if (!roomMessages[room]) {
+            roomMessages[room] = [];
+        }
+        roomMessages[room].push(messageData);
+        if (roomMessages[room].length > 200) {
+            roomMessages[room].splice(0, roomMessages[room].length - 200);
+        }
+        console.log(`💾 Stored message in room ${room}. Total messages in history:`, roomMessages[room].length);
         io.to(room).emit('chat-message', messageData);
         console.log(`Message sent to room ${room}: ${message}`);
     });
 
     socket.on('disconnect', () => {
         console.log(`Client disconnected: ${socket.id}`);
+
+        // Clean up user identification mapping
+        delete socketToUser[socket.id];
 
         // Remove user from all rooms they were in
         for (const [roomId, participants] of Object.entries(rooms)) {
@@ -178,10 +215,10 @@ io.on('connection', (socket: Socket) => {
                 // Notify remaining users in the room
                 socket.broadcast.to(roomId).emit('system-message', `A user has left the room`);
 
-                // Clean up empty rooms
+                // Clean up empty rooms (but keep message history)
                 if (participants.size === 0) {
                     delete rooms[roomId];
-                    console.log(`Deleted empty room: ${roomId}`);
+                    console.log(`Deleted empty room: ${roomId} (keeping message history)`);
                 }
             }
         }

